### DISCUSSION TITLE
Delegation should not be tied to a specific validator

### DISCUSSION THREAD

**Discussion by wehrv** [ava-labs/avalanchego#1777] [General]
Requiring delegators to choose a validator creates unnecessary friction in their decision making, and results in a loss of their potential reward if the validator chosen does not meet uptime requirements.  Choosing a specific validator should only be needed in limited circumstances, such as when the validator themselves, or those with close relationships to the validator, select a specific node for reasons irrelevant to the purpose of encouraging AVAX hodlers to lock up their assets for a specified time.

Delegators should be offered (at least) an *option* to choose to delegate to all compliant validators, with their reward being a specified percentage less than the validators themselves receive.  With the current minimum delegation expense being 2%, the fee could be set at perhaps 5%, since the delegators by choosing the "all validators" option now operate with the knowledge they cannot lose their potential reward if any given validator does not meet uptime requirements - however the fee could (eventually!) be set by governance decisions by validators, who collectively gain from this delegation option by receiving a reward slightly greater than the collective delegators.

**Comment by aphexmunky** [comment]
I think this might be best solved at the wallet level?

You're right that there's unnecessary friction in making a user choose a validator but it looks like core have taken a big leap in improving that UX. That's likely to improve and simplify over time.

Choosing to delegate against all validators is trickier, but even if you could i don't think you could guarantee the delegators reward

**Comment by wehrv** [comment]
Core's recent improvements are what made me think (well, reminded me) of this, though someone on the Discord/Twitter brought up the idea some time ago.  I think it'd be fairly easy (says he not writing the code) to implement, since the AVAX never leaves the wallet already, but is marked as validated/delegated with a specific node.  For an all-node validation, mark it against some special-case node (all zeros?), and at the delegation expiration time that special-case would be recognized and the uptime computations could be skipped.  The wallet would indeed need to support the special-case of an all-node delegation.

**Reply by StephenButtolph** [reply]
> I think it'd be fairly easy (says he not writing the code) to implement

It wouldn't be difficult

**Comment by StephenButtolph** [comment]
I think the biggest questions I'd have with this change is:

1. Would the ideal world here be to _replace_ the existing delegation mechanism with this?
2. Would it be possible to more scientifically define the new "delegation" rewards? Specially, just throwing a 5% reduction seems arbitrary (or at least the rational doesn't seem particularly analytical).

I suppose in this new world, the delegators would be locking up their funds to reduce the available supply that could be used to attack the network. Although that is sort of implying that the attackers aren't able to bribe the validators.

Total side note - I don't think this mechanism is really delegation anymore... It feels more like a treasury bond mechanism or something. (Someone would need to come up with a cool name)

**Reply by t-anyu** [reply]
I think this also has an opportunity to futher enhance decentralization if locked AVAX can be distributed more evenly across the network.

One way to circumvent the treasury bond mechanism is to do it like an automatic delegation selection feature. Where user defines parameters like locking period and fee. But then again, I guess a global delegation pool or treasury bond could make more sense in terms of innovative way of staking. Go figure.
